torch/onnx/_internal/fx/passes/functionalization.py

```diff
@@ -102,15 +103,26 @@ def _run(self, *args) -> torch.fx.GraphModule:
         module = _utils.wrap_graph_module_for_node_meta_preservation(self.module)
 
         functionalized_callable = self._functionalize(module)
-        fx_mode = "symbolic" if self.enable_dynamic_axes else "fake"
-
-        graph_module = proxy_tensor.make_fx(
-            functionalized_callable,
-            decomposition_table={},
-            tracing_mode=fx_mode,
-            _allow_non_fake_inputs=True,
-            _allow_fake_constant=self.allow_fake_constant,
-        )(*args)
+
+        fake_mode: Any = self.fake_mode
+        maybe_fake_args = self._maybe_fakefy_args(fake_mode, *args)
+        if fake_mode is not None:
+            # Using existing fake mode as context, signal `make_fx` that it does not need
+            # to create a new fake mode by passing tracing_mode as "real".
+            tracing_mode = "real"
```

Comment:
https://github.com/pytorch/pytorch/issues/105464 tracks fake mode + dynamic shape support which might change how `tracing_mode` and `fake_mode` interacts in near future.

I suggest a TODO to help finding this spot in the future when I revisit dyn shape+fake

Reply:
I can add one if it helps. But be sure to remove the TODO once we get more clarity on fake + dynamic.